samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/MyNotesActivity.java

```diff
@@ -69,40 +198,176 @@ protected void handleUserSignedOut() {
-                    getString(R.string.continote_url_to_edit_note_with_key, "someNoteKey"),
-                    getString(R.string.app_name_for_firebase_continue)
-            ).addOnSuccessListener(this, new OnSuccessListener<Void>() {
-                @Override
-                public void onSuccess(Void result) {
-                    showSnackbar(R.string.broadcast_to_continue_writing_note_in_chrome_successful);
-                }
-            }).addOnFailureListener(this, new OnFailureListener() {
-                @Override
-                public void onFailure(@NonNull Exception e) {
-                    showSnackbar(R.string.broadcast_to_continue_writing_note_in_chrome_failed);
-                }
-            });
+    public void handleWriteNoteButtonTapped(@Nullable View v) {
+        if (!currentUserIsSignedIn() || mNotesRef == null) {
+            // This should never happen, but just in case.
+            throw new AssertionError(
+                    "User must be signed in and their notes database reference must be non-null");
         }
+
+        // Try to add a new, empty Note to the Firebase Realtime Database for the current user.
+        addNoteToDatabase(new Note("", ""))
+                .addOnSuccessListener(
+                        MyNotesActivity.this,
+                        new OnSuccessListener<String>() {
+                            @Override
+                            public void onSuccess(String databaseKey) {
+                                // Open the Edit Note screen with this Note.
+                                // We do this so that the user may immediately edit the Note,
+                                // rather than having to wait for the ListView to update from
+                                // Firebase Realtime Database events and then manually tap the
+                                // list item for the Note.
+                                openEditScreenForNoteWithKey(databaseKey);
+                            }
+                        })
+                .addOnFailureListener(
+                        MyNotesActivity.this,
+                        new OnFailureListener() {
+                            @Override
+                            public void onFailure(Exception e) {
+                                showSnackbar(R.string.create_new_note_failed);
+                            }
+                        });
+    }
+
+    /**
+     * Attempts to asynchronously add a Note for the current user to the Firebase Realtime Database.
+     *
+     * This could be modified to go into the Note class, but this is the only place it is used,
+     * so this is sufficient for this sample app.
+     *
+     * @param note The Note to add to the database for the current user.
+     * @return A Task which, upon completion, signals whether or not the Note was successfully
+     * added to the database. Upon success, the Task provides the database key of the new Note.
+     */
+    @NonNull
```

Comment:
NonNull doesn't need to be explicitly defined anywhere. Java style (google's anyways) suggests its overly verbose: All variables should be treated as NonNull, and only variables that are explicitly Nullable are annotated as such.

(here and elsewhere)